pkg/virt-handler/device-manager/device_controller_test.go

```diff
@@ -194,12 +194,18 @@ var _ = Describe("Device Controller", func() {
 			}, 5*time.Second).Should(BeNumerically(">=", 1))
 		})
 
-		It("should remove all device plugins if permittedHostDevices is removed from the CR", func() {
+		It("Should remove device plugins if permittedHostDevices is removed from the CR", func() {
 			emptyConfigMap, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{})
 			Expect(emptyConfigMap.GetPermittedHostDevices()).To(BeNil())
 
-			initialDevices := []Device{plugin1, plugin2}
-			deviceController := NewDeviceController(host, maxDevices, permissions, initialDevices, emptyConfigMap, clientTest.CoreV1())
+			deviceController := NewDeviceController(host, maxDevices, permissions, []Device{}, emptyConfigMap, clientTest.CoreV1())
+
+			func() {
+				deviceController.startedPluginsMutex.Lock()
+				defer deviceController.startedPluginsMutex.Unlock()
```

Comment:
We don't need the lock since we did not run the controller yet

Reply:
I have dropped it.